source/authentication/custom-jwt.txt

```diff
@@ -101,106 +132,79 @@ Choose your preferred method below.
 
-         "config": {
-           "audience": [
-             "<JWT Audience>",
-           ],
-           "requireAnyAudience": <boolean>,
-         }
-         
-      .. list-table::
-        :header-rows: 1
-        :widths: 15 30
+.. list-table::
+   :header-rows: 1
+   :widths: 15 30
 
-        * - Field
-          - Description
+   * - Field
+     - Description
 
-        * - ``audience``
+   * - | :guilabel:`Signing Algorithm`
+       | *config.signingAlgorithm*
 
-          - Array of Strings. The audience or audiences expected
-            to be found in a client JWT. 
+     - The cryptographic method that the external system uses to sign
+       the JWT. Custom authentication supports JWTs signed using any of
+       the following algorithms:
 
-        * - ``requireAnyAudience``
+       - ``HS256``
+       - ``RS256``
+       
+   * - | :guilabel:`Signing Key`
+       | *secret_config.signingKeys*
 
-          - Boolean. If ``false``, valid JWTs must include all of the listed
-            audiences. If ``true``, valid JWTs must include at least one of
-            the listed audiences.
+     - A list of up to three :ref:`Secrets <app-secret>`
+       that each contain a signing key used by the third party 
+       authentication system to sign :abbr:`JWTs (JSON Web Tokens)`.
+       The key can only contain ASCII letters, numbers, 
+       underscores, and hyphens, and must be between 32 and 512 characters long. 
+       The following is a valid 256-bit signing key:
 
-.. _custom-jwt-authentication-configuration-verification-method:
+       .. code-block:: sh
+         :copyable: false
+ 
+         231a58b00632c9c4d8ac02b268ca4caf8dd48fd020e3dffa72666523d860988f
 
-Verification Method
-~~~~~~~~~~~~~~~~~~~
+       .. note::
+ 
+         If you are uncertain of what value to use, consider visiting a random 
+         key generator website, like `keygen.io <https://keygen.io/>`_, and 
+         using one of the generated 256-bit values. 
 
-The :guilabel:`Verification Method` configures how the provider
-determines which signing algorithm and signing keys the external
-authentication system must use to sign each JWT.
+.. tabs-realm-admin-interfaces::
 
-You can either :ref:`manually specify signing keys
-<custom-jwt-authentication-configuration-manually-signing-keys>`
-or :ref:`specify a JSON Web Key URI
-<custom-jwt-authentication-configuration-jwk-uri>`.
+   .. tab::
+      :tabid: ui
 
-.. _custom-jwt-authentication-configuration-manually-signing-keys:
+      1. Set the Signing Algorithm:
 
-Manually Specify Signing Keys
-`````````````````````````````
+         .. figure:: /images/custom-jwt-signing-algorithm.png
+             :alt: The Signing Keys configuration inputs
 
-You can manually configure the signing algorithm and specify one or more
-signing keys that the external authentication system may use to sign
-JWTs.
+      2. Create one or more signing keys that is used to sign the JWT. To do 
```

Comment:
This sentence is technically grammatically correct, but still awkward. What do you think about cutting some of it to streamline?
```suggestion
      2. Create one or more signing keys to sign the JWT. To do 
```